test/jdk/javax/swing/JFileChooser/CustomApproveButtonTest.java

```diff
@@ -0,0 +1,131 @@
+import javax.swing.JFileChooser;
+import javax.swing.JFrame;
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
+
+/*
+ * @test
+ * @bug 8299522
+ * @key headful
+ * @summary JFileChooser Test to check for FileChooser's Approve Button Text
+ *          when CUSTOM_DIALOG type is set.
+ * @run main CustomApproveButtonTest
+ */
+
+public class CustomApproveButtonTest {
+    private final String customApproveText;
+    private final String lookAndFeel;
+    private JFrame frame;
+
+    public static void main(String[] args) throws Exception {
+        String aquaLAF = "com.apple.laf.AquaLookAndFeel";
+        SwingUtilities.invokeAndWait(new Runnable() {
+            public void run() {
+                UIManager.LookAndFeelInfo[] lookAndFeel = UIManager.getInstalledLookAndFeels();
+                for (UIManager.LookAndFeelInfo look : lookAndFeel) {
+                    if(look.getClassName().equals(aquaLAF)) {
+                        continue;
+                    }
+                    new CustomApproveButtonTest(look.getClassName());
+                }
+            }
+        });
+        System.out.println("Test Pass!");
+    }
+
+    private CustomApproveButtonTest(String lookAndFeel) {
+        System.out.println("Testing Look & Feel : "+lookAndFeel);
+        frame = new JFrame("CustomApproveButtonTest");
+        this.lookAndFeel = lookAndFeel;
+
+        try {
+            UIManager.setLookAndFeel(lookAndFeel);
+        } catch (Exception e) {
+            fail("Failed to set ", e);
+        }
+
+        JFileChooser fileChooser = new JFileChooser();
+        fileChooser.setDialogType(JFileChooser.CUSTOM_DIALOG);
+        frame.add(fileChooser);
+        frame.setVisible(true);
+        frame.pack();
+
+        customApproveText = UIManager.getString("FileChooser.customApproveButtonText", fileChooser.getLocale());
+
+        if (customApproveText == null || customApproveText.length() == 0) {
+            fail("Cannot find Custom Approve Button text From UIManager Property!");
+            return;
+        }
+
+        JButton customApproveButton = findCustomApproveButton(fileChooser);
+
+        if (customApproveButton == null) {
+            fail("Cannot find Custom Approve Button in FileChooser!");
+            return;
+        }
+
+        if(frame != null) {
+            frame.dispose();
+        }
+
+    }
+
+    private JButton findCustomApproveButton(Container container) {
+        JButton result = null;
+
+        for (int i = 0; i < container.getComponentCount(); i++) {
+            Component c = container.getComponent(i);
+
+            if (c instanceof JButton && customApproveText.equals(((JButton) c).getText())) {
+                result = (JButton) c;
+                break;
+            }
+
+            if (c instanceof Container) {
+                JButton button = findCustomApproveButton((Container) c);
+
+                if (result == null) {
+                    result = button;
+                }
```

Comment:
What's the `result == null` for? 

Is it meant to not recursively call findCustomApproveButton() if a JButton result is found? Because then the break at line 107 resolves that I think.

If it's to only have a result for the first instance of a non-null result, then wouldn't adding a similar break after `result = button` be better so we don't have to recursively call findCustomApproveButton() if a result is already found?

Reply:
Updated.